### PR TITLE
signup with passport.js

### PR DESCRIPTION
**Pull request by Zoneam**

### CODE REVIEW

**Review by Dimavich — APPROVED** [review]

File: controllers/api/userRoutes.js [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
-module.exports = router;
\ No newline at end of file
+router.post(
+  '/',
+  authConfig.authenticate('local-signup', {
+    successRedirect: '/',
+    failureRedirect: '/login',
+  })
+
+  // async (req, res) => {
```
Comment (vasylynash) [inline controllers/api/userRoutes.js]: Remove commented code

File: public/js/createEvent.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const createEvent = async (event) => {
+  event.preventDefault();
+  const eventDateTime =
+    document.querySelector('#form4').value.trim() +
```
Comment (vasylynash) [inline public/js/createEvent.js]: IDs need to be specific

File: public/js/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+const createEvent = async (event) => {
+  event.preventDefault();
+  const eventDateTime =
+    document.querySelector('#form4').value.trim() +
```
Comment (vasylynash) [inline public/js/index.js]: IDs need to be specific

File: public/js/index.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+const createEvent = async (event) => {
```
Comment (vasylynash) [inline public/js/index.js]: Duplicate code from createEvent.js?

File: public/js/signup.js [hunk 5]
```diff
@@ -1,22 +1,43 @@
 const signupForm = async (event) => {
   event.preventDefault();
-  const email = document.querySelector('#email').value.trim(); // modal input for email must have id email
-  const username = document.querySelector('#username').value.trim(); // modal input for username must have id username
-  const password = document.querySelector('#password').value.trim(); // modal input for password must have id password
+  const email = document.querySelector('#registerEmail').value.trim();
+  const username = document.querySelector('#regUsername').value.trim();
+  const password = document.querySelector('#registerPassword').value;
+  const confirmPassword = document.querySelector('#confirmPassword').value;
+  const validEmail =
```
Comment (vasylynash) [inline public/js/signup.js:7]: Where is validEmail used?

File: public/js/signup.js [hunk 6]
```diff
@@ -1,22 +1,43 @@
...
+          alert('Failed to sign up!');
+        }
+      } else {
+        alert('Password must have 8 characters or more');
+        document.querySelector('#registerPassword').value = '';
+        document.querySelector('#confirmPassword').value = '';
+      }
     } else {
-      alert('Failed to sign up!');
+      alert('Passwords dont match');
```
Comment (vasylynash) [inline public/js/signup.js:34]: Typo in dont

**Review by vasylynash — CHANGES_REQUESTED** [review]

**Review by vasylynash — APPROVED** [review]

**Pull request merged**